chain/gen/genesis/util.go

```diff
@@ -42,3 +48,73 @@ func doExecValue(ctx context.Context, vm vm.Interface, to, from address.Address,
 
 	return ret.Return, nil
 }
+
+// CalculateMinerCreationDepositForGenesis calculates the deposit required for creating a new miner
+// during genesis block creation. This is used when API is not available yet.
+func CalculateMinerCreationDepositForGenesis(ctx context.Context, vm vm.Interface, store adt.Store) (abi.TokenAmount, error) {
+	nh, err := vm.Flush(ctx)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("flushing vm: %w", err)
+	}
+
+	nst, err := state.LoadStateTree(store, nh)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("loading new state tree: %w", err)
+	}
+	// Get reward actor state
+	rewardActor, err := nst.GetActor(reward.Address)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("loading reward actor: %w", err)
+	}
+
+	rewardState, err := reward.Load(store, rewardActor)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("loading reward actor state: %w", err)
+	}
+
+	// Get power actor state
+	powerActor, err := nst.GetActor(power.Address)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("loading power actor: %w", err)
+	}
+
+	powerState, err := power.Load(store, powerActor)
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("loading power actor state: %w", err)
+	}
+
+	networkQAPowerSmoothed, err := powerState.TotalPowerSmoothed()
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("getting network QA power smoothed: %w", err)
+	}
+
+	pledgeCollateral, err := powerState.TotalLocked()
+	if err != nil {
+		return big.Zero(), xerrors.Errorf("getting pledge collateral: %w", err)
+	}
+
+	// For genesis, use the initial reserved amount as default circulating supply
+	circSupply := types.BigInt{Int: buildconstants.InitialFilReserved}
```

Comment:
The internal calculation of MinerCreationDeposit in the Genesis miner directly uses `buildconstants.InitialFilReserved` as the circulating supply. I am not sure if this is a problem.

Reply:
I am also not sure.  But since this has no impact on the live networks this is the sort of thing that we can address when issues show up later.